.meta/_partials/fields/_compression_options.toml.erb

```diff
@@ -3,17 +3,32 @@
     "common" => true,
     "default" => "none",
     "description" => """\
-The compression strategy used to compress the encoded event data before \
+The compression algorithm used to compress the encoded event data before \
 transmission.\
 """
 }.merge(options || {}) -%>
+
 <%- enum = {
     "none" => "No compression.",
     "gzip" => "[Gzip][urls.gzip] standard DEFLATE compression."
 }.merge(enum || {}) -%>
 
 [<%= namespace %>.compression]
+type = "table"
+description = """\
+Configures the compression specific sink behavior.\
+"""
+
+[<%= namespace %>.compression.children.codec]
 <%= options.to_toml %>
 
-[<%= namespace %>.compression.enum]
+[<%= namespace %>.compression.children.codec.enum]
```

Comment:
Given that we already use the term _codec_ to refer to things like JSON encoding, I would recommend something other than that here. I suggest either _algorithm_ or _method_ instead (probably the latter since the former is harder to type).

Reply:
Agree with this. I prefer `algorithm`.